pyirf/interpolation.py

```diff
@@ -0,0 +1,270 @@
+from astropy.io import fits
+from scipy.interpolate import griddata
+
+# definition from lstchain.io.io copied here to avoid dependency
+dl2_params_lstcam_key = 'dl2/event/telescope/parameters/LST_LSTCam'
+
+
+def read_mean_pars_data(data_file, pars):
+    """
+    Reads a DL2 data file and extracts the average values
+    of the parameters for interpolation
+
+    Parameters
+    ----------
+    data_file: ``string``
+        path to the DL2 data file
+    pars: list of ``string``
+        list of parameters as they can be evaluated from DL2 file
+
+    Returns
+    -------
+    interp_pos: tuple
+        tuple of average values of requested parameters
+    """
+
+    # read in the data
+    interp_pos = []  # position for which to interpolate
+    data = pd.read_hdf(data_file, key=dl2_params_lstcam_key)
+    for par in pars:
+        # we use here eval function that is considered potentially dangerous
+        # as it can execute arbitrary code, however this is the eval
+        # function from pandas, that is very much limitted to just
+        # the columns read from the file and math operations
+        # so it should be safe here (and it adds a lot of flexibility)
+        val = np.mean(data.eval(par[1]))
+        interp_pos.append(val)
+
+    return tuple(interp_pos)
+
+
+def interpolate_effective_area(aeff_all, pars_all, interp_pars, min_effective_area=1. * u.Unit('m2'), method='linear'):
+    """
+    Takes a grid of effective areas for a bunch of different parameters
+    and interpolates (log) effective areas to given value of those parameters
+
+    Parameters
+    ----------
+    aeff_all: np.array of astropy.units.Quantity[area]
+        grid of effective area, of shape (n_grid_points, n_fov_offset_bins, n_energy_bins)
+    pars_all: np.array
+        list of parameters corresponding to aeff_all, of shape (n_grid_points, n_interp_dim)
+    interp_pars: np.array
+        values of parameters for which the interpolation is performed, of shape (n_interp_dim)
+    min_effective_area: astropy.units.Quantity[area]
+        Minimum value of effective area to be considered for interpolation
+    method: 'linear’, ‘nearest’, ‘cubic’
+        Interpolation method
+
+    Returns
+    -------
+    aeff_interp: astropy.units.Quantity[area]
+        Interpolated Effective area array with shape (n_energy_bins, n_fov_offset_bins)
+    """
+
+    _, n_fov_offset_bins, n_energy_bins = aeff_all.shape
+
+    # get rid of units
+    aeff_all = aeff_all.to('m2').value
+    min_effective_area = min_effective_area.to('m2').value
+
+    # remove zeros and log it
+    aeff_all[aeff_all < min_effective_area] = min_effective_area
+    aeff_all = np.log(aeff_all)
+
+    # interpolation
+    aeff_interp = np.empty((n_energy_bins, n_fov_offset_bins))
+    for i_th in range(n_fov_offset_bins):
+        for i_en in range(n_energy_bins):
+            aeff_interp[i_en, i_th] = griddata(pars_all, aeff_all[:, i_th, i_en], interp_pars, method=method)
```

Comment:
These loops are not neeed, since you can directly write:

```
aeff_interp = griddata(pars_all, aeff_all, interp_pars, method=method)
```

if my test with a multidimensional target function was correct.

Reply:
thx @maxnoe I implemented it this way (just with additional axis transposition that was needed) and also did the same with dispersion matrix

I did not realize that this will work because https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.griddata.html says that values should have shape (n,), and here obviously it is not a simple vector. Nevertheless when I run it over the dummy test it works and provide the same output, and running it over the example makes it much faster (from 140s to 0.013s for 0.9M of interpolations !)